cfn-resources/cluster/cmd/resource/mappings_test.go

```diff
@@ -79,3 +80,39 @@ func regionConfig(provider, region string) *[]admin.CloudRegionConfig {
 		ProviderName: &provider,
 	}}
 }
+
+func TestNewHardwareSpec(t *testing.T) {
+	testCases := map[string]struct {
+		expected string
```

Comment:
q: is there a reason to not use admin.HardwareSpec for expected value? Just thinking adding the marshaling aspect makes it harder to maintain and not seeing a clear benefit.

Reply:
More compact and tests the behavior of marshaling that the API call will use. But I'm open to changing it

Reply:
> tests the behavior of marshaling that the API call will use

I believe this falls more in the scope of the SDK implementation, we have control over how we populate the SDK model and have understanding that if a nil value is provided for a pointer it is not sent in the request. Don't see this too relevant so adjust as you see fit.

Reply:
choosing to keep the code as is. In this case we would be notified if the SDK would change unexpectedly.